### PR TITLE
 Add: 'Developer Files' section on download pages 

### PR DESCRIPTION
**Pull request by glx22**
I don't really know how the website works, but I think it would be nice to have an easy access to pdb files. I hope this change is enough.

### CODE REVIEW

**Comment by LordAro** [comment]
Alas, it is not. As best as I can tell, PDB files are not currently distributed at all (they don't appear in the manifest files at (e.g.) https://proxy.binaries.openttd.org/openttd-releases/1.10.0-beta1/manifest.yaml )

I'm not sure they're even built in the CI currently, you'd need to check yourself - 
https://github.com/OpenTTD/OpenTTD/blob/2be619ea88837c6293129841511816c105b507a3/azure-pipelines/templates/release.yml#L67-L103

But they're certainly excluded from the manifest generation script - https://github.com/OpenTTD/OpenTTD/blob/2be619ea88837c6293129841511816c105b507a3/azure-pipelines/manifest.sh#L41
You'd have to ask @TrueBrain why that is...

**Comment by glx22** [comment]
If I copy the `openttd-<version>-windows-win64.exe` URI and replace `exe` with `pdb.xz` I can download it. But indeed maybe website itself doesn't know they exist.

**Comment by TrueBrain** [comment]
Your patch won't do what you think it does :) `pdb`, as `dbg.deb` are on the ignore list when generating the manifest, which is needed to generate URLs.

They are on the ignore list, because people kept downloading them. That is, 30% of the people who downloaded the `zip` or `exe` also downloaded the `pdb`. Because people don't know what it is for.

So I am very much against putting them in the download list, with the other files. History shows that it is a bad solution :)

What might work, is some redesigning, where we put this in a new part on the downloads page, something like "Debug files" or what-ever. Here we can list the `pdb` and `dbg.deb`  files.

Either way, if you make a PR for this repository, please create and run the docker to check if it fixes what you think it does :) It is the whole reason this is a Docker .. so you can try it out on your local machine too :P Saves you going back and forth with assumptions :D

Going to close this PR for now; as what-ever solution we pick, it should be looking different :D Please do invest time in it, as it is worth fixing this properly!

**Comment by TrueBrain** [comment]
PS: yes, `pdb` and `dbg.deb` are published on the CDN, and if you know the URL, are free to download :)

**Comment by nielsmh** [comment]
Make it a "Tools for developers (not required to play the game)" download section, that also has grfcodec, catcodec, language compiler, etc.

**Comment by glx22** [comment]
I tested something by adding a ``debug-files:`` section in manifest, used to generate the ``Debug Files`` section in
website-debug-in-manifest

Of course if no ``debug-files`` section is found, the output is like before
website-debug-not-in-manifest

**Comment by LordAro** [comment]
I like it

Perhaps "Developer files" and move sources into it as well?

**Comment by TrueBrain** [comment]
Cool :D Should the documentation also be in there? And I guess the dbg-deb files should also be there. 

I personally wouldn't add the dropdown in the "developers" section, as that is just a bit silly. We use that for binaries, as it autodetects OSes and points people to the right files. No need to do that for developers :D

**Comment by glx22** [comment]
Ok removed dropdown, renamed "Debug Files" to "Developer Files".
In the screenshot only pdb are present because I added them manually in ``_download/openttd-releases/latest.md``. Presence in developer section will just depend on manifest file, files in `files` are shown in "download", files in `dev_files` are shown in "developer".

File: _layouts/download.html [hunk 1]
```diff
@@ -76,4 +76,36 @@ <h3>Download {{ page.name }}</h3>
...
         <script type="text/javascript">detectByUA(document.getElementById("download-combo-state"), "{{ page.base }}"); updateCombo(document.getElementById("download-data"));</script>
     </div>
+    {% if page.dev-files %}
+    <div class="section-header">
+        <div class="section-icon"></div>
+        <h3>Developer Files</h3>
+    </div>
+    <div class="section-item">
+        <div class="content">
+            <p>These files are not required to play.</p>
```
Comment (TrueBrain) [inline _layouts/download.html]: Think this can use a bit more text

**Comment by glx22** [comment]
Need to add `dbg.deb` to  `_data/download-descriptions.yml` but I don't know what to use as description :)

**Comment by TrueBrain** [comment]
Now we have to agree which files to move to the `Developer Files`, and we need to change the manifest generation script to do the right thing there. This last part is in the OpenTTD repository itself (as @LordAro pointed out).

File: _layouts/download.html (hunk 1 above)
Comment (nielsmh) [reply _layouts/download.html]: `These files are not required to play, and do not add anything to the game. They can be useful for programmers who want to diagnose crash bugs.`
Only point of confusion I can think of is that "crash" in TT context can also mean two vehicles colliding :D

**Comment by glx22** [comment]
Maybe this can go live now :)

**Pull request merged**